src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,138 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int count;
+    public static final int MAX_CAPACITY = 10;
+    private Object[] elements = new Object[MAX_CAPACITY];
     @Override
     public void add(T value) {
+        if (count >= elements.length) {
+            elements = grow();
+        }
+        elements[count] = value;
+        count++;
 
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexforAdd(index);
+        count++;
+        Object[] newArray = new Object[MAX_CAPACITY];
```

Comment:
The `newArray` should be initialized with the current size of `elements` plus one, not `MAX_CAPACITY`. This ensures that the array can accommodate the new element.